src/components/query-viewer/query-viewer.tsx

```diff
@@ -0,0 +1,332 @@
+    parametersValue = "[" + parametersValue + "]";
+
+    return parametersValue;
+  }
+
+  private getElements(): string {
+    let elementsValue = "";
+    const elements = Array.from(
+      document.getElementsByTagName("gx-query-viewer-element")
+    );
+    elements.forEach(ax => {
+      let elementValue = "{";
+      elementValue += ' "Name": "' + ax.name + '"';
+      elementValue += ' ,"Title": "' + ax.elementTitle + '"';
+      elementValue += ' ,"Visible": "' + ax.visible + '"';
+      elementValue += ' ,"Type": ' + ax.type + '"';
+      elementValue += ' ,"Axis": "' + ax.axis + '"';
+      elementValue += ' ,"Aggregation": "' + ax.aggregation + '"';
+      elementValue += ' ,"DataField": ' + ax.dataField + '"';
+      if (ax.axisOrderType) {
+        elementValue += ', "AxisOrder": { "Type": "' + ax.axisOrderType + '"';
+        elementValue +=
+          ax.axisOrderValues != null
+            ? ', "Values": [' + ax.axisOrderValues + "]"
+            : "";
+        elementValue += "}";
+      }
+      if (ax.filterType) {
+        elementValue += ', "Filter": { "Type": ' + ax.filterType + '"';
+        elementValue +=
+          ax.filterValues != null
+            ? ', "Values": [' + ax.filterValues + "]"
+            : "";
+        elementValue += "}";
+      }
+      if (ax.expandCollapseType) {
+        elementValue +=
+          ', "ExpandCollapse": { "Type": "' + ax.expandCollapseType + '"';
+        elementValue +=
+          ax.expandCollapseValues != null
+            ? ' "Values": [' + ax.expandCollapseValues + "]"
+            : "";
+
+        elementValue += "}";
+      }
+      //TODO add grouping values
+      const formats = Array.from(
+        ax.getElementsByTagName("gx-query-viewer-element-format")
+      );
+
+      formats.forEach(format => {
+        elementValue += ' ,"Format":  {';
+        elementValue += ' "Picture": "' + format.picture + '"';
+        elementValue += ' ,"Subtotals": "' + format.subtotals + '"';
+        elementValue += ' ,"CanDragToPages": "' + format.canDragToPages + '"';
+        elementValue += ' ,"Style": "' + format.formatStyle + '"';
+        elementValue += ' ,"TargetValue": "' + format.targetValue + '"';
+        elementValue += ' ,"MaximumValue": "' + format.maximumValue + '"';
+        elementValue += " }";
+        //TODO add fromat styles
+      });
+      elementValue += "}";
+      elementsValue += elementsValue != "" ? "," : "";
+      elementsValue += elementValue;
+    });
+    elementsValue = "[" + elementsValue + "]";
+    return elementsValue;
+  }
+
+  private getWidth(): string {
+    const computedStyle = getComputedStyle(this.element);
+    return computedStyle.width;
+  }
+
+  private getHeight(): string {
+    const computedStyle = getComputedStyle(this.element);
+    return computedStyle.height;
+  }
+
+  render() {
+    return (
+      <div>
+        <iframe
+          name="query_viewer"
+          width={this.getWidth()}
+          height={this.getHeight()}
```

Comment:
I think you can avoid calculating sizes by taking advantage of `display: flex; flex: 1;` in all the ancestors of the iframe and `flex: 1;` in the `iframe`.

Reply:
I can't because I need to send in the post the width & height, for the javascript calculate the size.

Reply:
Ok, understood 👍